exonum-java-binding/core/src/main/java/com/exonum/binding/core/service/Node.java

```diff
@@ -26,8 +26,13 @@
 /**
  * An Exonum node context. Allows to add transactions to Exonum network
  * and get a snapshot of the database state.
+ *
+ * <p>Once the node is <em>closed</em>, it can no longer be used. Methods of a closed Node
+ * will throw an {@link IllegalStateException} if invoked.
```

Comment:
do we need to throw ISE at `getPublicKey`?

Reply:
We can add it, but I think we can get away with no Exception)

Reply:
To do that in a concise way, we could use node() method to access the node (not a field load) and put the check there.